### PR TITLE
Download processor

### PR DESCRIPTION
**Pull request by NivLipetz**

### CODE REVIEW

File: app/models/runner.js [hunk 1]
```diff
@@ -137,12 +151,16 @@ async function writeProcessorFile(fileContent) {
     }
 }
 
-function injectPlugins(testFile, jobConfig) {
-    const metricsPluginName = jobConfig.metricsPluginName.toLowerCase();
-    const metricsAdapter = require(`../adapters/${metricsPluginName}Adapter`);
-    let asciiMetricsExportConfig = (Buffer.from(jobConfig.metricsExportConfig, 'base64').toString('ascii'));
-    let parsedMetricsConfig = JSON.parse(asciiMetricsExportConfig);
-    testFile.config.plugins = metricsAdapter.buildMetricsPlugin(parsedMetricsConfig, jobConfig);
+async function getProcessorPathIfExists(jobConfig, test) {
```
Comment (enudler) [inline app/models/runner.js]: i would simplify it by
1. when you download the 'file' you know it base 64 so just decode it
2. get rid of writeProcessorFile func, it seems like it's not really needed.

File: app/connectors/customJSConnector.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const requestSender = require('../helpers/requestSender'),
```
Comment (enudler) [inline app/connectors/customJSConnector.js]: most of this file is duplicated.

imo you should have two public function which just passes the url to new private function which executes the request with the headers and response logging

File: app/models/runner.js [hunk 3]
```diff
@@ -137,12 +151,16 @@ async function writeProcessorFile(fileContent) {
...
+    let localProcessorPath;
+    if (test['file_id']) {
```
Comment (enudler) [inline app/models/runner.js]: maybe add log saying it's deprecated? and put a link to processors docs?

File: app/models/runner.js (hunk 1 above)
Comment (NivLipetz) [reply app/models/runner.js]: done

File: app/connectors/customJSConnector.js (hunk 2 above)
Comment (NivLipetz) [reply app/connectors/customJSConnector.js]: done

File: app/models/runner.js (hunk 3 above)
Comment (NivLipetz) [reply app/models/runner.js]: done

**Pull request merged**